metricflow/sql_clients/databricks.py

```diff
@@ -137,7 +138,7 @@ def sql_engine_attributes(self) -> SqlEngineAttributes:
         return DatabricksEngineAttributes()
 
     @staticmethod
-    def params_or_none(bind_params: SqlBindParameters) -> Optional[Dict[str, str]]:
+    def params_or_none(bind_params: SqlBindParameters) -> Optional[Dict[str, SQLColumnType]]:
```

Comment:
This currently works by accident, I think:

https://github.com/databricks/databricks-sql-python/blob/main/src/databricks/sql/client.py#L301

MyPy isn't catching it because we aren't typechecking external libraries correctly, but as soon as we fix that issue this will break.

For most things it should work in their runtime as long as the parameter spec and value type match (e.g., the spec is `%(colname)d` for an `int` parameter value), because their parameter escaper takes in an `Any` and returns the escaped value true to type, so when they do `query_string % escaped_parameters` everything will be fine.

However, they don't support booleans:

https://github.com/databricks/databricks-sql-python/blob/main/src/databricks/sql/utils.py#L160-L174

This seems like a bummer. :(

Reply:
As discussed:
1. I'll put up some integration tests to make sure this works for all engines.
2. For any that don't support other types, I'll add engine-specific logic to convert to string before passing to the SQL connector.